Ginger/GingerCoreNET/RunLib/CLILib/CLIDynamicFile.cs

```diff
@@ -249,6 +257,44 @@ private void LoadSealightsDetailsFromJSON(GingerExecConfig gingerExecConfig)
             WorkSpace.Instance.Solution.SealightsConfiguration.SealightsReportedEntityLevel = (SealightsConfiguration.eSealightsEntityLevel)gingerExecConfig.SealightsDetails.SealightsEntityLevel;
             WorkSpace.Instance.Solution.SealightsConfiguration.SealightsAgentToken = gingerExecConfig.SealightsDetails.SealightsAgentToken;
         }
+        private void LoadGlobalVariableValues(GingerExecConfig gingerExecConfig)
+        {
+            try
+            {
+                foreach (GlobalVariable customizedVar in gingerExecConfig.GlobalVariables)
+                {
+                    VariableBase originalVar = DynamicExecutionManager.FindItemByIDAndName<VariableBase>(
+                                   new Tuple<string, Guid?>(nameof(VariableBase.Guid), customizedVar.VariableID),
+                                   new Tuple<string, string>(nameof(VariableBase.Name), customizedVar.VariableName),
+                                   WorkSpace.Instance.Solution.Variables);
+                    if (originalVar != null)
+                    {
+                        switch (originalVar.VariableType)
+                        {
+                            case "String":
```

Comment:
why hard coded strings ? can we use enum or typeof(VariableString)

Reply:
We have used Generic function to find variable using either by Id or name, it by default case object to VariableBase so its not possible get actual type of object, VariableType property is already defined in all variable classes to identify the type of object so using it in switch as switch does not support typeof(object)